File: Data Structures & Algorithms/anagram-groups/submission-2.rs
impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut map : HashMap<Vec<i32>, Vec<String>> = HashMap::new();

        for s in strs {
            let mut word = vec![0; 26];
            for c in s.chars() {
                word[(c as usize) - ('a' as usize)] += 1;
            }
            map.entry(word).or_default().push(s);
        }
        let mut result : Vec<Vec<String>> = Vec::new();
        for (_key, group) in map {
            result.push(group);
        }
        return result;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-2.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string res = "";
        for (string s : strs) {
            res += s;
            res += '\a';
        }
        return res;
    }

    vector<string> decode(string s) {
        vector<string> res = {};
        string word = "";
        for (char c : s) {
            if (c != '\a') {
                word += c;
            } else {
                res.push_back(word);
                word = "";
            }
            // res.push_back(word);
        }
        return res;
    }
};
